lib/matplotlib/tests/test_axes.py

```diff
@@ -1198,7 +1198,12 @@ def test_pcolorargs():
     x[0] = np.NaN
     with pytest.raises(ValueError):
         ax.pcolormesh(x, y, Z[:-1, :-1])
-    x = np.ma.array(x, mask=(x < 0))
+    # Maksing a nan might raise a warning
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        x = np.ma.array(x, mask=(x < 0))
+        # Some versions of numpy do not warn, but should have max 1 warning
+        assert len(w) <= 1
```

Comment:
I would just `with np.errstate(invalid="ignore"): ...` (or whatever works to suppress the warning).  Otherwise, the assert is really about numpy behavior, which we do not actually want to test.